source/geometries/GenericWLSFiber.cc

```diff
@@ -0,0 +1,298 @@
+    core_rad_  = iclad_rad_ - clad_thickness;
+    if (with_coating_) oclad_rad_ -= coating_thickness;
+  }
+}
+
+
+void GenericWLSFiber::DefineMaterials()
+{
+  // Fiber cladding materials to be used ...
+  // PMMA:       in single cladding, and for inner in muticladding
+  // FPethylene: for outer in muticladding
+
+  iclad_mat_ = materials::PMMA();
+  iclad_mat_->SetMaterialPropertiesTable(opticalprops::PMMA());
+
+  // If 2 claddings, defining the outer cladding material
+  if (doubleclad_) {
+    oclad_mat_ = materials::FPethylene();
+    oclad_mat_->SetMaterialPropertiesTable(opticalprops::FPethylene());
+  }
+
+  // If optical properties of coating are set explicitly, use them
+  if (with_coating_ && coating_optProp_) {
+    coating_mat_->SetMaterialPropertiesTable(coating_optProp_);
+  }
+
+  // If optical properties of core are set explicitly, use them
+  if (core_optProp_) {
+    core_mat_->SetMaterialPropertiesTable(core_optProp_);
+  }
+}
+
+
+void GenericWLSFiber::Construct()
+{
+  ComputeDimensions();
+
+  DefineMaterials();
+
+  if (isround_)  BuildRoundFiber();
+  else           BuildSquareFiber();
+}
+
+
+void GenericWLSFiber::BuildRoundFiber()
+{
+  // Pointer to the innermost logical volume defined at any moment.
+  G4LogicalVolume* innermost_logic = nullptr;
+
+  // Coating (only if it exists ...)
+  G4LogicalVolume* coating_logic;
+  if (with_coating_) {
+    G4String coating_name = name_ + "_WLS";
+    G4cout << "**** Building COATING " << coating_name << G4endl;
+    G4Tubs* coating_solid =
+      new G4Tubs(coating_name, 0., thickness_/2., length_/2., 0., 360.*deg);
+    coating_logic =
+      new G4LogicalVolume(coating_solid, coating_mat_, coating_name);
+
+    // Optical surface
+    G4OpticalSurface* coating_optSurf =
+      new G4OpticalSurface(coating_name + "_OPSURF", glisur, ground,
+                           dielectric_dielectric, .01);
+    new G4LogicalSkinSurface(coating_name + "_OPSURF", coating_logic,
+                             coating_optSurf);
+
+    GeometryBase::SetLogicalVolume(coating_logic);
+    innermost_logic = coating_logic;
+  }
+
+  // Outer Cladding (only if it exists ...)
+  G4LogicalVolume* oclad_logic;
+  if (doubleclad_) {
+    G4String oclad_name = name_ + "_OCLAD";
+    G4cout << "**** Building OCLAD " << oclad_name << G4endl;
+    G4Tubs* oclad_solid =
+      new G4Tubs(oclad_name, 0., oclad_rad_, length_/2., 0., 360.*deg);
+    oclad_logic =
+      new G4LogicalVolume(oclad_solid, oclad_mat_, oclad_name);
+
+    if (GeometryBase::GetLogicalVolume() == nullptr)
+      GeometryBase::SetLogicalVolume(oclad_logic);
+    else
+      new G4PVPlacement(nullptr, G4ThreeVector(0., 0., 0.), oclad_logic,
+                        oclad_name, innermost_logic, false, 0, false);
+
+    innermost_logic = oclad_logic;
+  }
+
+  // Inner Cladding (always built)
+  G4String iclad_name = name_ + "_ICLAD";
+  G4cout << "**** Building ICLAD " << iclad_name << G4endl;
```

Comment:
Do we need maybe a verbosity check here (and in the following G4couts?)

Reply:
Fixed in the latest commit!